src/domain/shared/primitives.ts

```diff
@@ -107,3 +107,27 @@ export const balanceAmountFromNumber = <T extends WalletCurrency>({
     currency,
   }
 }
+
+export const parseErrorMessageFromUnknown = (error: unknown): string => {
```

Comment:
Shouldn't this be applied at the bottom of the Error class hierarchy?

Reply:
I sync'd with @vindard and we created an `error-parsers` file